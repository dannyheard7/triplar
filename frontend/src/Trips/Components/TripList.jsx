import React from "react";
import {Link} from "react-router-dom";
import Moment from "moment";

import '../styles/trips.css';
import {getTrips} from "../utils/actions";
import { connect } from "react-redux";

export class TripList extends React.Component {
    componentDidMount() {
        this.props.dispatch(getTrips())
    }

    render() {
        return (
            <div>
                <h1>Your Trips</h1>
                <div className="trip-list">
                    {this.props.trips.map(this.createListItem)}
                </div>
            </div>
        )
    }

    createListItem(trip) {
        return (
            <div className="card trip-card" key={trip.id}>
                <h3 className="card-title"><Link to={"/trips/" + trip.id}>{trip.name}</Link></h3>
                <p className="card-text">
                    <span>{trip.locations && trip.locations.map((location, index) => (index ? ', ': '') + location.city.name)}</span><br />
                    {Moment(trip.startDate).format('Do MMMM') } - {Moment(trip.endDate).format('Do MMMM') }
                </p>
            </div>
        );
    }
}

TripList.defaultProps = {
    trips: []
};

const mapStateToProps = state => ({
  trips: state.trips.trips,
});

export default connect(mapStateToProps)(TripList)