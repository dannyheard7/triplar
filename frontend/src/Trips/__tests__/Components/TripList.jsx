import React from "react";
import {shallow} from "enzyme";

import TripList from "Trips/Components/TripList";


const faker = require('faker');

describe('<TripList />', () => {
    describe('with no trips prop',() => {
        it('displays no <ListGroupItem /> component', () => {
            const container = shallow(<TripList />);
            expect(container.find(".trip-card").length).toEqual(0);
        });
    });

    describe('with trips prop',() => {
        let trips = null;

        beforeEach(function() {
            trips = [{'id': faker.random.number(), 'name': faker.random.word(), 'locations': [faker.address.city()]}];
        });

        it('displays a <ListGroupItem /> component for a single trip', () => {
            const container = shallow(<TripList trips={trips} />);
            expect(container.find(".trip-card").length).toEqual(trips.length);
        });

        it('updates on new trip props', () => {
            const container = shallow(<TripList trips={trips}/>);
            expect(container.find(".trip-card").length).toEqual(trips.length);

            let newTrips = [{'id': faker.random.number(), 'name': faker.random.word(), 'locations': [faker.address.city()]},
                {'id': faker.random.number(), 'name': faker.random.word(), 'locations': [faker.address.city()]}];
            container.setProps({trips: newTrips});
            expect(container.find(".trip-card").length).toEqual(newTrips.length);
        });
    });
});