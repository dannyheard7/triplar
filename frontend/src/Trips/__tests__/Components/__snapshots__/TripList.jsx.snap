// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripList /> renders a correctly 1`] = `
<div
  className="trip-list"
/>
`;
